File: code/pollen/app/src/main/kotlin/net/sprout/pollen/schemas/Enums.kt
package net.sprout.pollen.schemas

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class NodeType {
    @SerialName("rhizome") RHIZOME,
    @SerialName("pollen") POLLEN,
    @SerialName("meristem") MERISTEM
}

@Serializable
enum class Mode {
    @SerialName("normal") NORMAL,
    @SerialName("conservative") CONSERVATIVE,
    @SerialName("alert") ALERT
}

@Serializable
enum class ActionType {
    @SerialName("WATER_A") WATER_A,
    @SerialName("WATER_B") WATER_B,
    @SerialName("WATER_BOTH") WATER_BOTH,
    @SerialName("SKIP") SKIP,
    @SerialName("DEFER") DEFER,
    @SerialName("ALERT") ALERT,
    @SerialName("BLOCK") BLOCK,
    @SerialName("SHUTDOWN") SHUTDOWN
}

@Serializable
enum class ContradictionType {
    @SerialName("sensor_vs_vision") SENSOR_VS_VISION,
    @SerialName("policy_expired") POLICY_EXPIRED,
    @SerialName("deposit_inconsistent") DEPOSIT_INCONSISTENT,
    @SerialName("flow_vs_pump") FLOW_VS_PUMP,
    @SerialName("weather_conflict") WEATHER_CONFLICT,
    @SerialName("tank_underflow") TANK_UNDERFLOW
}

@Serializable
enum class Severity {
    @SerialName("info") INFO,
    @SerialName("warning") WARNING,
    @SerialName("critical") CRITICAL
}

@Serializable
enum class WeatherProvenance {
    @SerialName("local_station") LOCAL_STATION,
    @SerialName("ferried") FERRIED,
    @SerialName("forecast_api") FORECAST_API
}

File: code/shared/schemas/enums.py
"""Enums compartidos entre nodos."""

from __future__ import annotations

from enum import Enum


class Mode(str, Enum):
    NORMAL = "normal"
    CONSERVATIVE = "conservative"
    ALERT = "alert"


class ActionType(str, Enum):
    WATER_A = "WATER_A"
    WATER_B = "WATER_B"
    WATER_BOTH = "WATER_BOTH"
    SKIP = "SKIP"
    DEFER = "DEFER"
    ALERT = "ALERT"
    BLOCK = "BLOCK"
    SHUTDOWN = "SHUTDOWN"


class ContradictionType(str, Enum):
    SENSOR_VS_VISION = "sensor_vs_vision"
    POLICY_EXPIRED = "policy_expired"
    DEPOSIT_INCONSISTENT = "deposit_inconsistent"
    FLOW_VS_PUMP = "flow_vs_pump"
    WEATHER_CONFLICT = "weather_conflict"
    TANK_UNDERFLOW = "tank_underflow"


class Severity(str, Enum):
    INFO = "info"
    WARNING = "warning"
    CRITICAL = "critical"


class WeatherProvenance(str, Enum):
    LOCAL_STATION = "local_station"
    FERRIED = "ferried"
    FORECAST_API = "forecast_api"


class PatchOperation(str, Enum):
    REPLACE = "replace"
    ADD = "add"
    REMOVE = "remove"


class DeltaPriority(str, Enum):
    LOW = "low"
    NORMAL = "normal"
    HIGH = "high"
    CRITICAL = "critical"
